From c6ca6500d94ba456b14704f82b73588cc823022a Mon Sep 17 00:00:00 2001
From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Sun, 16 Aug 2026 14:14:37 +0200
Subject: [PATCH 18/20] refactor(devicetree-firmware): make looping over fw_dir
 top-level loop

Upstream-status: https://github.com/dracut-ng/dracut/pull/2598

Make the looping over $fw_dir the top-level loop. This is a preparation
patch for adding support to install_generic() to add soc specific
firmwares (qcom/<soc>/qupv3fw.elf).
---
 modules.d/70devicetree-firmware/module-setup.sh | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/modules.d/70devicetree-firmware/module-setup.sh b/modules.d/70devicetree-firmware/module-setup.sh
index 100e027a..16b9e00f 100755
--- a/modules.d/70devicetree-firmware/module-setup.sh
+++ b/modules.d/70devicetree-firmware/module-setup.sh
@@ -46,10 +46,10 @@ install_hostonly() {
 # generic install (hostonly / generic use completely different approaches)
 install_generic() {
     local _fw_files=()
-    # ATM only qcom WoA laptops need this
-    for _soc in qcom/sc8280xp qcom/x1e80100; do
-        # shellcheck disable=SC2154 # fw_dir is set by dracut.sh
-        for _fwdir in $fw_dir; do
+    # shellcheck disable=SC2154 # fw_dir is set by dracut.sh
+    for _fwdir in $fw_dir; do
+        # add laptop model specific firmwares for qcom WoA laptops
+        for _soc in qcom/sc8280xp qcom/x1e80100; do
             # add '*' after mbn, elf for .gz, etc. compression
             for _fw in "$_fwdir/$_soc"/*/*/*.mbn* "$_fwdir/$_soc"/*/*/*.elf*; do
                 [ -f "$_fw" ] || continue
-- 
2.55.0

